src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,400 @@
+    { rintro ⟨b, hb₁, hb₂⟩,
+      rw (encodable.decode₂_is_partial_inv _ _) at hb₁,
+      rwa ← encodable.encode_injective hb₁ } },
+
+  rw [summable.has_sum_iff, this, ← tsum_Union_decode₂],
+  { exact v.of_empty },
+  { rw hg₃, change summable ((λ i, v (g i)) ∘ encodable.encode),
+    rw function.injective.summable_iff encodable.encode_injective,
+    { exact (v.has_sum_of_disjoint_Union hg₁ hg₂).summable },
+    { intros x hx,
+      convert v.of_empty,
+      simp only [Union_eq_empty, option.mem_def, not_exists, mem_range] at ⊢ hx,
+      intros i hi,
+      exact false.elim ((hx i) ((encodable.decode₂_is_partial_inv _ _).1 hi)) } }
+end
+
+lemma measure_Union [encodable β] {f : β → set α}
+  (hf₁ : ∀ i, measurable_set (f i)) (hf₂ : pairwise (disjoint on f)) :
+  v (⋃ i, f i) = ∑' i, v (f i) :=
+(has_sum_measure_Union hf₁ hf₂).tsum_eq.symm
+
+lemma of_union {A B : set α}
+  (h : disjoint A B) (hA : measurable_set A) (hB : measurable_set B) :
+  v (A ∪ B) = v A + v B :=
+begin
+  rw [union_eq_Union, measure_Union, tsum_fintype, fintype.sum_bool, cond, cond],
+  exacts [λ b, bool.cases_on b hB hA, pairwise_disjoint_on_bool.2 h]
+end
+
+lemma of_diff {A B : set α} (hA : measurable_set A) (hB : measurable_set B)
+  (h : A ⊆ B) : v A + v (B \ A) = v B :=
+begin
+  rw [← of_union disjoint_diff hA (hB.diff hA), union_diff_cancel h],
+  apply_instance,
+end
+
+lemma of_diff' {M : Type*} [add_comm_group M]
+  [topological_space M] [t2_space M] {v : vector_measure α M}
+  {A B : set α} (hA : measurable_set A) (hB : measurable_set B)
+  (h : A ⊆ B) : v (B \ A) = v B - (v A) :=
+begin
+  rw [← of_diff hA hB h, add_sub_cancel'],
+  apply_instance,
+end
+
+lemma of_Union_nonneg {M : Type*} [topological_space M]
+  [ordered_add_comm_monoid M] [order_closed_topology M]
+  {v : vector_measure α M} (hf₁ : ∀ i, measurable_set (f i))
+  (hf₂ : pairwise (disjoint on f)) (hf₃ : ∀ i, 0 ≤ v (f i)) :
+  0 ≤ v (⋃ i, f i) :=
+(v.of_disjoint_Union hf₁ hf₂).symm ▸ tsum_nonneg hf₃
+
+lemma of_Union_nonpos {M : Type*} [topological_space M]
+  [ordered_add_comm_monoid M] [order_closed_topology M]
+  {v : vector_measure α M} (hf₁ : ∀ i, measurable_set (f i))
+  (hf₂ : pairwise (disjoint on f)) (hf₃ : ∀ i, v (f i) ≤ 0) :
+  v (⋃ i, f i) ≤ 0 :=
+(v.of_disjoint_Union hf₁ hf₂).symm ▸ tsum_nonpos hf₃
+
+lemma of_nonneg_disjoint_union_eq_zero {s : signed_measure α} {A B : set α}
+  (h : disjoint A B) (hA₁ : measurable_set A) (hB₁ : measurable_set B)
+  (hA₂ : 0 ≤ s A) (hB₂ : 0 ≤ s B)
+  (hAB : s (A ∪ B) = 0) : s A = 0 :=
+begin
+  rw of_union h hA₁ hB₁ at hAB,
+  linarith,
+  apply_instance,
+end
+
+lemma of_nonpos_disjoint_union_eq_zero {s : signed_measure α} {A B : set α}
+  (h : disjoint A B) (hA₁ : measurable_set A) (hB₁ : measurable_set B)
+  (hA₂ : s A ≤ 0) (hB₂ : s B ≤ 0)
+  (hAB : s (A ∪ B) = 0) : s A = 0 :=
+begin
+  rw of_union h hA₁ hB₁ at hAB,
+  linarith,
+  apply_instance,
+end
+
+lemma summable_measure {μ : measure α} [hμ : finite_measure μ]
```

Comment:
this one shouldn't be in the `vector_measure` namespace, as it deals with usual measures. It should probably even be in a different file, on measures.

Reply:
I moved it to `measure_space` and changed the name to `summable_measure_to_real` since I think that's more descriptive?